src/parameterized_tendencies/gravity_wave_drag/non_orographic_gravity_wave.jl

```diff
@@ -726,24 +502,173 @@ function non_orographic_gravity_wave_forcing(
             level_end,
         )
 
+        # update gravity wave forcing
         @. uforcing = uforcing + u_waveforcing
         @. vforcing = vforcing + v_waveforcing
 
     end
     return nothing
 end
 
-# calculate the intermittency factor eps -> assuming constant Δc.
+# Using column_accumulate function, calculate the gravity wave forcing at each point.
+function waveforcing_column_accumulate!(
```

Comment:
I'm not keen on nested anonymous functions like this. Can we somehow break this giant function into smaller ones?

Reply:
I've refactored part of the wave source calculation into smaller function calls. Does this approach work for you? Breaking down the unrolled_reduce function, which is a big part of the whole function, isn’t easy because it directly uses many externally passed parameters. If we avoid using anonymous functions, we’ll have to keep passing those parameters in the reduce operation, which might make the function less elegant.

Reply:
`wave_source` looks great, thanks for updating that. It would be nice if it's possible to do something similar with the block inside the `unrolled_reduce(Val(nc), (mask, FT1(0.0))) do (mask, fm), (n)` block, but I understand if it's not possible.